GVRf/Framework/src/org/gearvrf/scene_objects/GVRVideoSceneObject.java

```diff
@@ -167,8 +189,16 @@ public MediaPlayer getMediaPlayer() {
      * @param mediaPlayer
      *            An Android {@link MediaPlayer}
      */
-    public void setMediaPlayer(MediaPlayer mediaPlayer) {
-        mVideo.setMediaPlayer(mediaPlayer);
+    public void setMediaPlayer(final MediaPlayer mediaPlayer) {
+        if (mVideo == null) {
+            getGVRContext().runOnGlThread(new Runnable() {
+                @Override
+                public void run() {
+                    mVideo.setMediaPlayer(mediaPlayer);
```

Comment:
In line with all the other mVideo null checks, is there a need for one more while on the gl thread?